doc/sphinx/language/core/sections.rst

```diff
@@ -65,9 +65,7 @@ usable outside the section as shown in this :ref:`example <section_local_declara
    This can be used to define a term incrementally,
    in particular by relying on the :tacn:`refine` tactic.
    See :ref:`proof-editing-mode`.
-   The proof should be terminated with :cmd:`Defined`. Using :cmd:`Qed` in
-   Let is deprecated. To get the same effect (hiding the body of the
-   proof), add the :attr:`clearbody` attribute to the Let.
+   The proof should be terminated with :cmd:`Defined`.
```

Comment:
Should be something like
```suggestion
   If the proof is terminated with :cmd:`Qed`, it is used to define an opaque side definition (not local to the section), then the side definition provides the body for the section local constant.
```
In other words
~~~coq
Let foo : type.
Proof. tactics. Qed.
~~~
in a section is equivalent to 
~~~coq
(* name "foo_subproof" auto generated to avoid conflicts *)
Local Lemma foo_subproof : type.
Proof. tactics. Qed.
#[clearbody] Let foo := foo_subproof.
~~~
(possibly with some differences in polymorphic universe handling but maybe any such differences could be considered bugs)

Reply:
This is already perfectly explained just below.